pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -366,9 +381,15 @@ def get_treeinfo_data(remote, remote_url):
             treeinfo["repositories"].update({directory: str(sub_repo.pk)})
             path = f"{repodata}/"
             new_url = urlpath_sanitize(remote_url, path)
-            with tempfile.TemporaryDirectory("."):
-                repodata_exists = get_repomd_file(remote, new_url)
-            if repodata_exists:
+
+            try:
+                with tempfile.TemporaryDirectory("."):
+                    get_repomd_file(remote, new_url)
+            except ClientResponseError as exc:
+                if exc.status == 404:
```

Comment:
This is now (I think) equivalent to what it was doing before, but to be honest I'm still not sure whether it's "correct" to go looking for a repo that was declared in the treeinfo and ignore it if it isn't there.

Reply:
"correct" is such an ambiguouos word when it comes to distribution-trees :)  We have a ton of heuristics in the kickstart code, that are the direct response to adapting to the realities "in the wild". @goosemania  and @ipanova have been burned way worse by this than I have tho, and would know better'n me.

Reply:
I'll take a note to run this past them once they're back from PTO